### DISCUSSION TITLE
Github auth with Personal Access Token?

### DISCUSSION THREAD

**Discussion by klorinczi** [gitextensions/gitextensions#10774] [General]
I try to push branch with Git Extensions to GitHub, but when pushing the change it displays a Github pop-up window to login. 

I created a new access token on Github.com / Settings Developer settings / Personal access tokens / Tokens (classic). I checked repo checkboxes (Full control of private repositories), nothing else. 
I added the token I got into Git Extensions / Tools / Settings / Plugins / GitHub / Personal Access Token (Global for all repositories).

Unfortunately this does not authenticate, still displays a Github pop-up window to login.

**Comment by vbjay** [comment]
The access token has nothing to do with remotes. The token allows the plugin to do things like create pull requests and fork repos and such.

**Comment by klorinczi** [comment]
I have to push changes to the Github repo.
But pops up the Github login page.

Is this caused because Personal Access Token has not enough permissions?

**Reply by vbjay** [reply]
THE OATH TOKEN HAS NOTHING to do with pushing or pulling.

**Reply by klorinczi** [reply]
Then what I do wrong, that Git Extensions pops up a Github login window when pushing?

**Reply by vbjay** [reply]
Note the settings node is in plugins section. Note the GitHub menu. Note how GitHub uses a rest API to do things like fork and create pull request.

**Reply by vbjay** [reply]
Login to approve the push. Is your remote https?  Lookup how git credential manager works.

**Reply by gerhardol** [reply]
https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-authentication-to-github#authenticating-with-the-command-line

https/GCM or ssh keys must be setup.

**Reply by klorinczi** [reply]
I use HTTPS remote. 
I also installed Github for command line.
Used following command to login:
```
c:\git\GitHub CLI>gh auth login
? What account do you want to log into? GitHub.com
? You're already logged into github.com. Do you want to re-authenticate? Yes
? What is your preferred protocol for Git operations? HTTPS
? Authenticate Git with your GitHub credentials? Yes
? How would you like to authenticate GitHub CLI? Paste an authentication token
Tip: you can generate a Personal Access Token here https://github.com/settings/tokens
The minimum required scopes are 'repo', 'read:org', 'workflow'.
? Paste your authentication token: 
- gh config set -h github.com git_protocol https
? Configured git protocol
? Logged in as user
```

I also have SSH key setup in Github.

**Reply by vbjay** [reply]
If you have ssh setup then configure ssh use in GE. Use putty or open ssh. I myself prefer putty.

**Reply by klorinczi** [reply]
I found a trivial/lame problem based on your comments.
The repo had HTTPS URL set, not SSH.

After setting up an SSH address for the repo, the push worked fine as it should.

Thank you @vbjay , @gerhardol for your help & patience!

**Reply by vbjay** [reply]
Been there done that. :D

**Reply by vbjay** [reply]
Small tip. You can set https pull url and ssh push if you want.
